src/fromager/candidate.py

```diff
@@ -28,12 +28,14 @@ def __init__(
         url: str,
         extras: typing.Iterable[str] | None = None,
         is_sdist: bool | None = None,
+        build_tag: int | None = None,
```

Comment:
I recommend to use a new, user-defined type for build tag. Build tags are not limited to numbers. They can be any string. pip sorts build tags by converting the leading digits to an integer and then sorting by tuple `(int(digits), remainder)`. In the future we can use the build tag to annotate the build flavor, e.g. `0.cpu_ubi9`.

Reply:
Should settings also return the `BuildTag` type then for now it could just be `(int(digits), "")`?